4/README.md

```diff
@@ -0,0 +1,346 @@
+
+Properties for event reporting purposes:
+* *rel change*, *abs change*, *archive rel chagne*, *archive abs chagne*,
+  describing the threshold values for relative and absolute change
+  in Attribute's *read value* above which
+  a *CHANGE* event or *ARCHIVE* event MUST be reported.
+  It MUST be either a valid numerical value
+  or a pair of valid numerical values separated by a `,`.
+  If one value is specified, it MUST be used for both negative
+  and positive change.
+  If two values are specified, the first MUST be used for netative change
+  and the second MUST be used for positive change.
+  It MUST be defined only for Attributes with numerical *data type*,
+* *period*, *archive period*, describing the minimum time period
+  in milliseconds after which a *PERIODIC* or *ARCHIVE* event MUST be reported,
+  regardless of Attribute's *read value*. If *period* is not specified,
+  a default value of 1000 ms is used,
+  It MUST be defined only for Attributes with numerical *data type*.
+
+See [Attribute events section](#attribute-events) for more details.
+
+```abnf
+rel-change = change
+abs-change = chagne
+
+archive-rel-change = change
+archive-abs-change = chagne
+
+number = [ "-" ] 1*DIGIT [ "." ] *DIGIT
+change = number [ "," number ]
+
+period = [ "-" ] 1*DIGIT
+archive-period = period
+```
+
+### Attribute runtime parameters
+
+At given point in time an Attribute MUST have associated:
+* *quality*, an enumeration describing the state *read value*
+  (one of *VALID*, *INVALID*, *ALARM*, *CHANGING*, *WARNING*),
+* *read value*, an object representing the value of the Attribute.
+  It MUST conform to *data format* and *data type*,
+* *read dim x*, an integer describing the number of data elements in
+  *read value* in X dimension.
+  If *data format* is *SCALAR*, it MUST be either 0 or 1.
+  If *data format* is *SPECTRUM* or *IMAGE*, it MUST be between 0 and
+  *max dim x*.
+* *read dim y*, an integer describing the number of data elements in
+  *read value* in Y dimension.
+  If *data format* is *SCALAR* or *SPECTRUM*, it MUST be 0.
+  If *data format* is *IMAGE*, it MUST be between 0 and *max dim y*.
+
+Additionally, if *writable* is *WRITE* or *READ_WRITE*,
+at given point in time an Attribute MUST have associated:
+* *set value*, an object representing the value set to the Attribute.
+  It MUST conform to *data format* and *data type*,
+* *write dim x*, an integer describing the number of data elements in
+  *set value* in X dimension,
+* *write dim y*, an integer describing the number of data elements in
+  *set value* in Y dimension.
+
+
+### Attribute aliases
+
+A *full attribute name* consists of *device name* and *attribute name*
+separated by "/" character.
+
+See [RFC 2/Device](/2) for the definition of `<device-name>`.
+
+An Attribute MAY have associated *alias*, a string which can be used
+in place of *full attribute name* to address the Attribute.
+An Attribute MUST NOT have more than one *alias*.
```

Comment:
I think this is a limitation of the current system with jive but I don't see why we could not have several aliases for the same attribute. How did you come to this conclusion? Did you find something in the doc or in the source code or was it simply by trying to define an attribute alias with jive?

Reply:
This is how it is implemented in the database:
https://github.com/tango-controls/TangoDatabase/blob/master/DataBase.cpp#L6150-L6155

Upon alias definition, you clear all existing aliases. Also see:

```
In [2]: db = tango.Database()

In [3]: db.put_attribute_alias('sys/tg_test/1/long_scalar', 'first_alias_for_test_long_scalar')

In [4]: db.get_attribute_alias_list('*')
Out[4]: DbDatum(name = '*', value_string = ['first_alias_for_test_long_scalar', 'rw_scalar_double-x', '', 'debian8_attr_alias'])

In [5]: db.put_attribute_alias('sys/tg_test/1/long_scalar', 'second_alias_for_test_long_scalar')

In [6]: db.get_attribute_alias_list('*')
Out[6]: DbDatum(name = '*', value_string = ['second_alias_for_test_long_scalar', 'rw_scalar_double-x', '', 'debian8_attr_alias'])
```

Reply:
You're right indeed! What you wrote is correct! I just don't understand why we have this limitation for the moment. I thought it might be useful in some situations to have several aliases pointing to the same attribute.
Maybe this is not needed.